src/test/java/com/google/sps/CalendarServletTest.java

```diff
@@ -0,0 +1,139 @@
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.sps.data.Event;
+import com.google.sps.data.User;
+import com.google.sps.Trip;
+import com.google.sps.servlets.CalendarServlet;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import static org.mockito.Mockito.*;
+
+@RunWith(JUnit4.class)
+public final class CalendarServletTest {
+
+  private CalendarServlet calendarServlet;
+
+  // class constants
+  private static final String INPUT_DESTINATION = 
+      "4265 24th Street San Francisco, CA, 94114";
+  private static final String INPUT_DATE = "2020-07-15";
+  private static final String TEST_NAME = "testName";
+  private static final String EMPIRE_ADDRESS = "20 W 34th St, New York, NY 10001";
+  private static final int HALF_HOUR = 30;
+  private static final LocalDateTime DEF_START_TIME = 
+      LocalDateTime.of(LocalDate.parse(INPUT_DATE), LocalTime.of(10, 0));
+  private static final String EMAIL = "test123@gmail.com";
+
+  // Cosntants to represent different Trip attributes.
+  private static final String TRIP_NAME = "Trip to California";
+  private static final String DESTINATION_NAME = "California";
+  private static final String IMAGE_SRC =
+    "https://lh3.googleusercontent.com/p/AF1QipM7tbCZOj_5SOft9cYgI7un3bmieieqvdYkCPT5=s1600-w400";
+  private static final String TRIP_DAY_OF_TRAVEL = "2020-02-29";
+
+  private final LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+
+  @Before
+  public void initCalendarServlet() {
+    calendarServlet = new CalendarServlet();
+  }
+
+   @Before
+  public void setUp() {
+    helper.setUp();
+  }
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
+  }
+
+  @Test
+  public void testWriteEventsCorrectly() throws Exception {
```

Comment:
I think you're missing some test cases here e.g. no logged-in user, no trips returned, multiple trips returned, multiple events, multiple trip days

Reply:
Unless the user goes to .../calendar.html then there will always be a logged in user. there will always be a trip returned because once Chris's CL gets merged the URL will contain a tripID. I can add multiple events for a tripDay but for the MVP we are limiting the Trip to a single day

Reply:
So I looked into datastore and it does not allow you to give entities the same key so it is not possible for me to make that test. I did add a test that has multiple tripDays and events